src/GraphQL/Execution/SubscriptionExecutionStrategy.cs

```diff
@@ -0,0 +1,310 @@
+                        };
+
+                        return await ProcessErrorAsync(context, node, exception).ConfigureAwait(false);
+                    });
+        }
+
+        /// <summary>
+        /// Processes data from the event source via <see cref="IObserver{T}.OnNext(T)"/> and
+        /// returns an <see cref="ExecutionResult"/>.
+        /// <br/><br/>
+        /// Override this method to mutate <see cref="ExecutionContext"/> as necessary, such
+        /// as changing the <see cref="ExecutionContext.RequestServices"/> property to a scoped instance.
+        /// </summary>
+        protected virtual async Task<ExecutionResult> ProcessDataAsync(ExecutionContext context, ExecutionNode node, object? value)
+        {
+            var result = new ExecutionResult(context);
+
+            try
+            {
+                // "clone" the node and set the source
+                // overwrite the 'node' variable here so it is picked up by exception handling code below
+                // and will contain the source from this data event
+                node = BuildSubscriptionExecutionNode(node.Parent!, node.GraphType!, node.Field, node.FieldDefinition, node.IndexInParentNode, value!);
+
+                if (context.Listeners != null)
+                {
+                    foreach (var listener in context.Listeners)
+                    {
+                        await listener.BeforeExecutionAsync(context)
+                            .ConfigureAwait(false);
+                    }
+                }
+
+                if (context.Errors.Count > 0)
+                {
+                    result.AddErrors(context.Errors);
+                    return result;
+                }
+
+                // Execute the whole execution tree and return the result
+                await ExecuteNodeTreeAsync(context, node).ConfigureAwait(false);
+
+                try
+                {
+                    if (context.Listeners != null)
+                    {
+                        foreach (var listener in context.Listeners)
+                        {
+                            await listener.AfterExecutionAsync(context)
+                                .ConfigureAwait(false);
+                        }
+                    }
+                }
+                finally
+                {
+                    // Set the execution node's value to null if necessary
+                    var dataIsNull = node.PropagateNull();
+
+                    // Return the result
+                    result.Executed = true;
+                    if (!dataIsNull || node.FieldDefinition.ResolvedType is not NonNullGraphType)
+                    {
+                        result.Data = new RootExecutionNode(null!, null)
+                        {
+                            SubFields = new ExecutionNode[]
+                            {
+                                node,
+                            }
+                        };
```

Comment:
I don't understand/remember that part. Why `|| node.FieldDefinition.ResolvedType is not NonNullGraphType` ? `ExecutionStrategy` has no such condition:
```
  // After the entire node tree has been executed, get the values
            object? data = rootNode.PropagateNull() ? null : rootNode;

            return new ExecutionResult(context)
            {
                Executed = true,
                Data = data,
            };
```

Reply:
I think it works differently because it's not the root node.  I agree that is was not obvious why this code was needed in exactly this manner, and puzzled me a bit, but I was sure to test all scenarios and it is correct.

See tests:
- Widget_WithErrors
- Widget_WithErrors_Nullable
- Widget_Null
- Widget_Null_Nullable

Reply:
Null handling in this respect was one of the bugs I attempted to fix. Previously there was a comment in the code that simply said that fields were treated as nullable even if they were not, with a note to fix it.  This code should now properly meet the GraphQL spec for nullability.

I can provide more details of my reasoning if you have any questions.  I did not investigate why this particular code had to be added here where it did not in ExecutionStrategy.